### DISCUSSION TITLE
Type of Spatie\MediaLibrary\Models\Media::$casts must be array (as in class Illuminate\Database\Eloquent\Model)

### DISCUSSION THREAD

**Discussion by elmarzouguidev** [spatie/laravel-medialibrary#2800] [Q&A]
php 7.4 problem with typing

**Comment by freekmurze** [comment]
The tests indicate that everything is working as expected. If you find that this error is caused by the package, feel free to PR a fix.

**Comment by elmarzouguidev** [comment]
I  just add '**array**' to typage and i solve the problem :

namespace Spatie\MediaLibrary\MediaCollections\Models;

class Media extends Model implements Responsable, Htmlable

befor : 

    protected $casts = [
        'manipulations' => 'array',
        'custom_properties' => 'array',
        'generated_conversions' => 'array',
        'responsive_images' => 'array',
    ];

after : 

    protected array  $casts = [
        'manipulations' => 'array',
        'custom_properties' => 'array',
        'generated_conversions' => 'array',
        'responsive_images' => 'array',
    ];

**Reply by elmarzouguidev** [reply]
@freekmurze  what do you think ??

**Reply by freekmurze** [reply]
So, this is a documentation error? If so, feel free to send a PR to the docs.

**Comment by elmarzouguidev** [comment]
@freekmurze  THE PROBLEM is that $casts in Illuminate\Database\Eloquent\Model are **typed** with **array**  : 
protected array  $casts = [];
so the iteration make problem **but** this problem is thrown in PHP 7.4
and "laravel/framework": "^8.65"

 I don't know what about php 7.3 or 7.2 ...
